base-files: fix Linksys upgrade, restore config step

trinidude4 opened the pull request:
It appears that the refactor of the upgrade process for NAND devices resulted in the nand_do_upgrade_success step not being called for devices using the linksys.sh script. As a result, configuration was not preserved over sysupgrade steps.

This restores the preservation of configs for kirkwood, and mvebu/cortexa9 devices and fixes an error in ipq40xx and ipq806x, using the linksys.sh script.

Fixes: e25e6d8
Fixes: #12298
Fixes: 8634c1080d5033e8cf0069ee7447821726232b95
Fixes: 2715aff5df836c1d3eb8cbf5afdb0c197902ee4b

Signed-off-by: Michael Trinidad <trinidude4@hotmail.com>

trinidude4 committed:
kirkwood: fix Linksys upgrade, restore config step

It appears that the refactor of the upgrade process for NAND devices
resulted in the nand_do_upgrade_success step not being called for
devices using the linksys.sh script. As a result, configuration was
not preserved over sysupgrade steps.

This restores the preservation of configs for kirkwood devices using the
linksys.sh script.

Fixes: e25e6d8e5407 ("base-files: fix and clean up nand sysupgrade code")
Fixes: #12298
Signed-off-by: Michael Trinidad <trinidude4@hotmail.com>

anomeome commented:
Tested on mvebu / mamba, rango
Issue extent on mamba and is resolved with PR, rango never experienced issue.

trinidude4 commented:
@hauke I have a branch conflict now.  2715aff was merged in but it has the same typo as 8634c10.  It might call nand_do_upgrade_failure which doesn't exist instead of nand_do_upgrade_failed.  The typo could be fixed, or I would propose keeping the code as in this PR because it matches closer to the code in package/base-files/files/lib/upgrade/nand.sh.

hauke commented:
Please adapt the code like it was done in target/linux/ipq40xx/base-files/lib/upgrade/linksys.sh

Ansuel commented:
@trinidude4 please split this to 3 commit, one for mvebu, one for kirkwood and the other that fix the typo

And please one reference for fixes tag

```
Fixes: e25e6d8e5407119746cba624244a5ac1a5cdbc83
Fixes: #12298 
```

@anomeome would like to give a tested-by tag?

trinidude4 committed:
mvebu: cortexa9: fix Linksys upgrade, restore config step

It appears that the refactor of the upgrade process for NAND devices                                                                                                                            resulted in the nand_do_upgrade_success step not being called for
devices using the linksys.sh script. As a result, configuration was
not preserved over sysupgrade steps.

This restores the preservation of configs for mvebu/cortexa9 devices using the
linksys.sh script.

Fixes: e25e6d8e5407 ("base-files: fix and clean up nand sysupgrade code")
Signed-off-by: Michael Trinidad <trinidude4@hotmail.com>

trinidude4 committed:
base-files: fix Linksys upgrade, restore config step

It appears that the refactor of the upgrade process for NAND devices                                                                                                                            resulted in the nand_do_upgrade_success step not being called for
devices using the linksys.sh script. As a result, configuration was
not preserved over sysupgrade steps.

This corrects a typo in the call of nand_do_upgrade_failed for ipq40xx
and ipq806x devices using the linksys.sh script.

Fixes: 8634c1080d50 ("ipq40xx: Fix Linksys upgrade, restore config step")
Fixes: 2715aff5df83 ("ipq806x: Fix Linksys upgrade, restore config step")
Signed-off-by: Michael Trinidad <trinidude4@hotmail.com>

trinidude4 commented:
@Ansuel I split this into 3 commits.  Is it ok to keep it all in one PR?
